src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/InstanceNameTest/InstanceNameTest.cs

```diff
@@ -84,48 +84,96 @@ public static void ConnectManagedWithInstanceNameTest(bool useMultiSubnetFailove
+#if NETCOREAPP
+        [ActiveIssue("27981")] // DataSource.InferNamedPipesInformation is not initializing InstanceName field
+        [ConditionalTheory(nameof(IsSPNPortNumberTestForNP))]
+        [InlineData("")]
+        [InlineData("44444")] // Named Instance Sql Server Port will be setup in the pipeline to 44444 as well
+        public static void PortNumberInSPNTestForNP(string port)
         {
-            string connStr = DataTestUtility.TCPConnectionString;
-            // If config.json.SupportsIntegratedSecurity = true, replace all keys defined below with Integrated Security=true 
+            string connectionString = DataTestUtility.NPConnectionString;
+            SqlConnectionStringBuilder builder = new(connectionString);
+
+            if (!string.IsNullOrWhiteSpace(port))
+                builder.DataSource = $"{builder.DataSource},{port}";
+
+            PortNumberInSPNTest(builder.ConnectionString);
+        }
+
+        [ConditionalTheory(nameof(IsSPNPortNumberTestForTCP))]
+        [InlineData("")]
+        [InlineData("44444")] // Named Instance Sql Server Port will be setup in the pipeline to 44444 as well
+        public static void PortNumberInSPNTestForTCP(string port)
+        {
+            string connectionString = DataTestUtility.TCPConnectionString;
+            SqlConnectionStringBuilder builder = new(connectionString);
+
+            if (!string.IsNullOrWhiteSpace(port))
+                builder.DataSource = $"{builder.DataSource},{port}";
+
+            PortNumberInSPNTest(builder.ConnectionString);
+        }
+#endif
+
+        private static void PortNumberInSPNTest(string connectionString)
+        {
+            // If config.json.Supports IntegratedSecurity = true, replace all keys defined below with Integrated Security=true 
             if (DataTestUtility.IsIntegratedSecuritySetup())
             {
                 string[] removeKeys = { "Authentication", "User ID", "Password", "UID", "PWD", "Trusted_Connection" };
-                connStr = DataTestUtility.RemoveKeysInConnStr(DataTestUtility.TCPConnectionString, removeKeys) + $"Integrated Security=true";
+                connectionString = DataTestUtility.RemoveKeysInConnStr(connectionString, removeKeys) + $"Integrated Security=true";
+            }
+
+            SqlConnectionStringBuilder builder = new(connectionString);
+
+            string hostname = "";
+            string instanceName = "";
+            int port = -1;
+
+            // Named pipe protocol data source does not support port number
+            string dataSource = builder.DataSource.ToUpper();
+            if ((dataSource.Contains(@"\MSSQL$") && builder.DataSource.ToUpper().Contains(@"\SQL\QUERY") && dataSource.Contains(",")) ||
+               (dataSource.Contains(@"NP:") && dataSource.Contains(",")))
+            {
+                port = -2;
             }
+            Assert.False(port == -2, "Named pipe protocol in data source does not support port number.");
```

Comment:
Move this into Line 138. And do Assert.Fail

Reply:
Moved Assert to line 138 and used Assert.Fail as suggested.